From 80ada5d41d46d6e12b10d3b29219996c7dbf449f Mon Sep 17 00:00:00 2001
From: Andrei Gherzan <andrei@resin.io>
Date: Wed, 3 Aug 2016 16:39:32 +0100
Subject: [PATCH 1/2] artik710_raptor.h: Set CONFIG_ROOT_PART to 2

Upstream-Status: Inappropiate [Configuration]

Signed-off-by: Andrei Gherzan <andrei@resin.io>
---
 include/configs/artik710_raptor.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/artik710_raptor.h b/include/configs/artik710_raptor.h
index aad8a42..253375f 100644
--- a/include/configs/artik710_raptor.h
+++ b/include/configs/artik710_raptor.h
@@ -299,7 +299,7 @@
 #define CONFIG_ROOT_DEV		0
 #define CONFIG_BOOT_PART	1
 #define CONFIG_MODULE_PART	2
-#define CONFIG_ROOT_PART	3
+#define CONFIG_ROOT_PART	2
 
 #define CONFIG_EXTRA_ENV_SETTINGS					\
 	"fdt_high=0xffffffffffffffff\0"					\
-- 
2.9.0

